python/dgl/nn/pytorch/conv/gatconv.py

```diff
@@ -170,21 +170,23 @@ def __init__(
         self.feat_drop = nn.Dropout(feat_drop)
         self.attn_drop = nn.Dropout(attn_drop)
         self.leaky_relu = nn.LeakyReLU(negative_slope)
-        if bias:
-            self.bias = nn.Parameter(
-                th.FloatTensor(size=(num_heads * out_feats,))
-            )
-        else:
-            self.register_buffer("bias", None)
         if residual:
             if self._in_dst_feats != out_feats * num_heads:
                 self.res_fc = nn.Linear(
-                    self._in_dst_feats, num_heads * out_feats, bias=False
+                    self._in_dst_feats, num_heads * out_feats, bias=bias
                 )
             else:
                 self.res_fc = Identity()
+                if bias:
+                    self.bias = nn.Parameter(
+                        th.FloatTensor(size=(num_heads * out_feats,))
+                    )
         else:
             self.register_buffer("res_fc", None)
+            if bias:
+                self.bias = nn.Parameter(
+                    th.FloatTensor(size=(num_heads * out_feats,))
+                )
```

Comment:
Let's simplify this branch to make it more readable. How about:
```python
self.has_res_fc = False
self.has_bias = False
if residual:
    if self._in_dst_feats != out_feats * num_heads:
        self.res_fc = nn.Linear(
            self._in_dst_feats, num_heads * out_feats, bias=bias
        )
        self.has_res_fc = True
    else:
        self.res_fc = Identity()

if bias and not self.has_res_fc:
    self.bias = nn.Parameter(
        th.FloatTensor(size=(num_heads * out_feats,))
    )
    self.has_bias = True
```

Reply:
@jermainewang Thank you for your suggestion. I have updated the code but I am concerned that the names of the flags might be misleading: eg. the flag has_bias is set to false when actually bias is present but inside nn.Linear. What do you think of changing eg. has_bias for has_separate_bias and has_res_fc for has_linear_res?

Reply:
Or `has_explicit_bias`?